test/testsupport/stubs/EssttpBackend.scala

```diff
@@ -0,0 +1,235 @@
+        |        "markedAsInsolvent" : false,
+        |        "isLessThanMinDebtAllowance" : false,
+        |        "isMoreThanMaxDebtAllowance" : false,
+        |        "disallowedChargeLocks" : false,
+        |        "existingTTP" : false,
+        |        "exceedsMaxDebtAge" : false,
+        |        "eligibleChargeType" : false,
+        |        "missingFiledReturns" : false
+        |      },
+        |      "chargeTypeAssessment" : [ {
+        |        "taxPeriodFrom" : "2020-08-13",
+        |        "taxPeriodTo" : "2020-08-14",
+        |        "debtTotalAmount" : 300000,
+        |        "disallowedChargeLocks" : [ {
+        |          "chargeId" : "A00000000001",
+        |          "mainTrans" : "mainTrans",
+        |          "mainTransDesc" : "mainTransDesc",
+        |          "subTrans" : "subTrans",
+        |          "subTransDesc" : "subTransDesc",
+        |          "outstandingDebtAmount" : 100000,
+        |          "interestStartDate" : "2017-03-07",
+        |          "accruedInterestToDate" : 15.97,
+        |          "chargeLocks" : {
+        |            "paymentLock" : {
+        |              "status" : false,
+        |              "reason" : ""
+        |            },
+        |            "clearingLock" : {
+        |              "status" : false,
+        |              "reason" : ""
+        |            },
+        |            "interestLock" : {
+        |              "status" : false,
+        |              "reason" : ""
+        |            },
+        |            "dunningLock" : {
+        |              "status" : false,
+        |              "reason" : ""
+        |            }
+        |          }
+        |        } ]
+        |      } ]
+        |    },
+        |    "taxId" : {
+        |      "value" : "123/456"
+        |    }
+        |  },
+        |  "sessionId" : "IamATestSessionId",
+        |  "createdAt" : "2022-05-18T14:04:03.461"
+        |}
+        |""".stripMargin
+  }
+
+  object CanPayUpfront {
+
+    def updateCanPayUpfront(journeyId: JourneyId, canPayUpfrontScenario: Boolean): StubMapping =
+      stubFor(
+        post(urlPathEqualTo(s"/essttp-backend/journey/${journeyId.value}/update-can-pay-upfront"))
+          .withRequestBody(equalTo(canPayUpfrontScenario.toString))
+          .willReturn(
+            aResponse()
+              .withStatus(200)
+          )
+      )
+
+    def findJourneyAfterUpdateCanPayUpfront(canPayUpfront: CanPayUpfront): StubMapping = {
+      val additionalPayloadInfo: (String, Boolean) =
+        if (canPayUpfront.value) {
+          ("Yes", true)
+        } else {
+          ("No", false)
+        }
+      stubFor(
+        get(urlPathEqualTo(findByLatestSessionIdUrl))
+          .willReturn(aResponse()
+            .withStatus(200)
+            .withBody(jsonBody(additionalPayloadInfo)))
+      )
+    }
+
+    val jsonBody: ((String, Boolean)) => String = (tuple: (String, Boolean)) =>
```

Comment:
you could make this easier to read by making it a `def like:
```scala
def jsonBody(blah: String, blahblah: Boolean): String = ...
```
with suitable names for the input parameters